tracim/views/core_api/session_controller.py

```diff
@@ -75,54 +50,31 @@ def logout(self, context, request: TracimRequest, hapic_data=None):
         return
 
     @hapic.with_api_doc()
-    @hapic.handle_exception(
-        NotAuthentificated,
-        http_code=HTTPStatus.UNAUTHORIZED
-    )
-    @hapic.output_body(
-        UserSchema(),
-    )
+    @hapic.handle_exception(NotAuthentificated, HTTPStatus.UNAUTHORIZED)
+    @hapic.output_body(UserSchema(),)
     def whoami(self, context, request: TracimRequest, hapic_data=None):
         """
         Return current logged in user or 401
         """
         app_config = request.registry.settings['CFG']
-        return UserInContext(
-            user=request.current_user,
-            dbsession=request.dbsession,
+        uapi = UserApi(
+            request.current_user,
+            session=request.dbsession,
             config=app_config,
         )
+        user = uapi.get_current_user()  # User
+        return uapi.get_user_with_context(user)
 
     def bind(self, configurator: Configurator):
 
         # Login
-        configurator.add_route(
-            'login',
-            os.path.join(BASE_API_V2, 'sessions', 'login'),
-            request_method='GET'
-        )
-        configurator.add_view(
-            self.login,
-            route_name='login',
-        )
+        configurator.add_route('login', '/sessions/login', request_method='POST')  # nopep8
```

Comment:
would be nice to defin routes directly as view method decorators. is it possible ?

Reply:
complicated.

Reply:
we previously decide to not rely on pyramid default decorator. Working with those (view pyramid decorator) doesn't simplify a lot code (as we already need to set route in another place) and class decorator doesn't work well with hapic , see https://github.com/algoo/hapic/issues/42